File: 프로그래머스/2/87946. 피로도/피로도.js
function solution(k, dungeons) {
    let answer = 0;
    let result = [];
    let visited = Array.from({length: dungeons.length}, () => false)
    
    const DFS = (count, k) => {
        result.push(count);
        
        for (let i=0; i<dungeons.length; i++) {
            const [cur, need] = dungeons[i];
            if (k >= cur && !visited[i]) {
                visited[i] = 1;
                DFS(count + 1, k - need);
                visited[i] = 0;
            }
        }   
    }
    DFS(0, k)
    answer = Math.max(...result)
    return answer;
}

File: 프로그래머스/2/87946. 피로도/피로도.py
answer = 0

def dfs(k, cnt, dungeons, visited):
    global answer
    if cnt > answer:
        answer = cnt

    # 최소 피로도, 소모 피로도
    for i in range(len(dungeons)):
        [minimum, consume] = dungeons[i]
        if not visited[i] and k >= minimum:
            visited[i] = True
            dfs(k-consume, cnt+1, dungeons, visited) 
            visited[i] = False
                
def solution(k, dungeons):
    global answer
    visited = [False] * len(dungeons)
    
    dfs(k, 0, dungeons, visited)
    
    return answer